dvc/repo/diff.py

```diff
@@ -85,3 +50,57 @@ def _exists(output):
     }
 
     return ret if any(ret.values()) else {}
+
+
+def _paths_checksums(repo):
+    """
+    A dictionary of checksums addressed by relpaths collected from
+    the current tree outputs.
+
+    To help distinguish between a directory and a file output,
+    the former one will come with a trailing slash in the path:
+
+        directory: "data/"
+        file:      "data"
+    """
+
+    return dict(_output_paths(repo))
+
+
+def _output_paths(repo):
+    repo_tree = RepoTree(repo, stream=True)
+    on_working_tree = isinstance(repo.tree, LocalTree)
+
+    def _exists(output):
+        if on_working_tree:
+            return output.exists
+        return True
+
+    def _to_path(output):
+        return (
+            str(output)
+            if not output.is_dir_checksum
+            else os.path.join(str(output), "")
```

Comment:
@pmrowla, is this intended? The output will have both directory and it's contents, right? Eg:
```
Added:
    data/
    data/1.txt
```

Reply:
Yes, it's intended that it should show both the directory and the contents